Utilities/Services/SitemapBuilder.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using QCVault.Utilities.Entities;
+
+namespace QCVault.Utilities.Services
+{
+    public class SitemapBuilder
+    {
+        private readonly XNamespace NS = "http://www.sitemaps.org/schemas/sitemap/0.9";
+
+        private readonly List<SitemapUrl> urls;
+
+        public SitemapBuilder()
+        {
+            urls = new List<SitemapUrl>();
+        }
+
+        public void AddUrl(string url, DateTime? modified = null, ChangeFrequency? changeFrequency = null, double? priority = null)
+        {
+            urls.Add(new SitemapUrl()
+            {
+                Url = url,
+                Modified = modified,
+                ChangeFrequency = changeFrequency,
+                Priority = priority,
+            });
+        }
+
+        public override string ToString()
+        {
+            var sitemap = new XDocument(
+                new XDeclaration("1.0", "utf-8", "yes"),
+                new XElement(NS + "urlset",
+                    from item in urls
+                    select CreateItemElement(item)
+                    ));
+
+            return sitemap.ToString();
+        }
+
+        private XElement CreateItemElement(SitemapUrl url)
+        {
+            var itemElement = new XElement(NS + "url", new XElement(NS + "loc", url.Url.ToLower()));
+
+            if (url.Modified.HasValue)
+            {
+                itemElement.Add(new XElement(NS + "lastmod", url.Modified.Value.ToString("yyyy-MM-ddTHH:mm:ss.f") + "+00:00"));
+            }
```

Comment:
I think "modified" just shouldn't be included right now. We don't have good tracking of when something is actually modified, and I'd rather provide no data than incorrect data.